xagon-react-app/src/rendering/TriangleMesh/State/MeshStateRotating.ts

```diff
@@ -188,67 +195,87 @@ class MeshStateRotating extends IMeshState {
                 //   b.setScale(scalingBone);
                 // });
 
-                // #SHIFT_ROTATION_NODE
-                const deltaShift =
-                  firstTriangle_AdjacentEdgeCenter_TriangleCenter.length() /
-                  secondTriangle_AdjacentEdgeCenter_TriangleCenter.length();
-                console.log(deltaShift);
+                // // #SHIFT_ROTATION_NODE
+                // const deltaShift =
+                //   firstTriangle_AdjacentEdgeCenter_TriangleCenter.length() /
+                //   secondTriangle_AdjacentEdgeCenter_TriangleCenter.length();
+                // console.log(deltaShift);
 
-                scalingNodeFirstTriangle.position =
-                  scalingNodeFirstTriangle.position.scale(1 / deltaShift);
+                // scalingNodeFirstTriangle.position =
+                //   scalingNodeFirstTriangle.position.scale(1 / deltaShift);
 
-                switch (direction) {
+                const rotationAxis =
+                  this.firstEdges[firstTriangleMeshFlipEdgeIndex];
+
+                switch (this.direction) {
                   case 1: {
-                    flipNodeFirstTriangle.rotationQuaternion =
-                      Quaternion.RotationAxis(
-                        firstEdges[firstTriangleMeshFlipEdgeIndex],
-                        firstTriangleFlipAngle,
-                      );
-                    this.nextState = new MeshStateIdle({
-                      triangleMesh: this.triangleMesh,
-                      scene: this.scene,
-                    });
+                    const rotationAnglePerFrame =
+                      this.getRotationAnglePerFrame(rotationDownAngle);
+                    // flipNodeFirstTriangle.rotationQuaternion =
+                    //   Quaternion.RotationAxis(rotationAxis, rotationDownAngle);
+
+                    // this.nextState = new MeshStateIdle({
+                    //   triangleMesh: this.triangleMesh,
+                    //   scene: this.scene,
+                    // });
+                    // console.log('FIRST', (rotationDownAngle * 180) / Math.PI);
+                    // console.log('ANGLE', this.angle);
 
                     // #ROTATION_TRIANGLE1
-                    // if (this.angle >= Math.abs(firstTriangleFlipAngle)) {
-                    //   this.nextState = new MeshStateIdle({
-                    //     triangleMesh: this.triangleMesh,
-                    //     scene: this.scene,
-                    //   });
-                    // } else {
-                    //   flipNodeFirstTriangle.rotationQuaternion =
-                    //     Quaternion.RotationAxis(
-                    //       firstEdges[firstTriangleMeshFlipEdgeIndex],
-                    //       this.angle * direction,
-                    //     );
-                    // }
+                    if (this.angle > rotationDownAngle) {
+                      flipNodeFirstTriangle.rotationQuaternion =
```

Comment:
cmq in linea generale:
- nel costruttore calcoliamo il quaternione di partenza e quello finale e li salviamo come variabili membro.
- nell'update calcoliamo il valore di amount da passare alla slerp  che ci restituisce il quaternione interpolato in quell'istante temporale.